### PR TITLE
Fixes #37 : Turn analyze into a polling request

### LINKED ISSUE #37: Turn analyze into a polling request
Description
Change analyze route to be a polling request to avoid timeout issue.
This happens on github codespaces when after a minute github codespaces cancels the request

Tasks
- [ ] Make the data analyzis asynchrone
- [ ] If the data is not yet analyzed, return a Retry-after json
- [ ] If the data is available, return the data
Acceptance Criteria
- [ ] Frontend hosted on github codespaces doesnt get timedout
- [ ] The data is still sent to the frontend

### PR DESCRIPTION
**Pull request by Youllou**
Made a function get_data that get the data.
Called it on a thread in the analyze route
While the data hasnt been retrieved the route returns a Retry-after:10

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -53,14 +53,21 @@ def upload_image():
...
-@app.route('/analyze', methods=['GET'])
-def analyze_document():
-    document = document_storage.get_document()
-    if not document:
-        return "No documents to analyze", 400
+
+
+def get_data(document):
     
+    global SHOULD_GET_DATA
```
Comment (snakedye) [inline app.py:67]: Should be defined in the global scope.

File: app.py [hunk 2]
```diff
@@ -53,14 +53,21 @@ def upload_image():
         # Add image to document storage
         document_storage.add_image(file_path)
         
+        
+        global SHOULD_GET_DATA 
+        SHOULD_GET_DATA = False
```
Comment (snakedye) [inline app.py:57]: Unsure why its in so many places.

File: app.py [hunk 3]
```diff
@@ -72,13 +79,37 @@ def analyze_document():
...
+    
+    document = document_storage.get_document()
+    if not document:
+        return "No documents to analyze", 400
+    
+    
+    
+    print(f"data from route pov : {data}")
+    if data == "":
+        if not SHOULD_GET_DATA :
```
Comment (snakedye) [inline app.py:96]: Wouldn't this lock any concurrent request to the backend?

**Review by snakedye — COMMENTED** [review]
On hold until I test Ollama. This might be unnecessary if its fast enough.

File: app.py (hunk 1 above)
Comment (Youllou) [reply app.py:67]: even if in global scope, must specify the global in function to access the global var

File: app.py (hunk 2 above)
Comment (Youllou) [reply app.py:57]: tried to make it the most logical possible, might have over used it

File: app.py (hunk 3 above)
Comment (Youllou) [reply app.py:96]: Yes it does,
I believe there are other issues with concurrent request to the backend
Might all be solved with user accounts later

**Comment by snakedye** [comment]
I closed issue #37. If I don't get a response by the end of the day this will be closed @Youllou.

**Comment by Youllou** [comment]
I confirm that it can be closed

**Pull request closed without merging**